src/Workspaces/Core/Portable/Storage/StorageOptions.cs

```diff
@@ -17,8 +17,15 @@ internal static class StorageOptions
 
         public const string OptionName = "FeatureManager/Storage";
 
-        public static readonly Option<StorageDatabase> Database = new(
-            OptionName, nameof(Database), defaultValue: StorageDatabase.SQLite);
+        public static readonly Option<StorageDatabase> Database = new(OptionName, nameof(Database), defaultValue: StorageDatabase.SQLite);
+
+        /// <summary>
+        /// Option that can be set in certain scenarios (like tests) to indicate that the client expects the DB to
+        /// succeed at all work and that it should not ever gracefully fall over.  Should not be set in normal host
+        /// environments, where it is completely reasonable for things to fail (for example, if a client asks for a key
+        /// that hasn't been stored yet).
+        /// </summary>
+        public static readonly Option<bool> DatabaseMustSucceed = new(OptionName, nameof(DatabaseMustSucceed), defaultValue: false);
```

Comment:
yes.  that's another good place.  where would i make such a change?

Reply:
Ideally I'd like to see options reset tied in here:
https://github.com/dotnet/roslyn/blob/4c67d21d48861bd4007fe37e57cd46289cfef344/src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstance.cs#L236-L237

Reply:
And I should say: if we want to make that bit in a separate PR I'm OK with that, the reason being I could imagine flipping it might expose some flakiness if we do have bugs we're unaware of. Like make a separate PR and run tests a bunch of times first before calling it good.

Reply:
yeah, i'll do a different PR.